ilc/client/handlePageTransaction.spec.js

```diff
@@ -0,0 +1,255 @@
+import chai from 'chai';
+import sinon from 'sinon';
+import html from 'nanohtml';
+
+import handlePageTransaction, {
+    slotCanBe,
+} from './handlePageTransaction';
+
+describe('handle page transaction', () => {
+    const locationHash = 'i-am-location-hash';
+
+    const slots = {
+        id: 'slots',
+        appendSlots: () => document.body.appendChild(slots.ref),
+        removeSlots: () => document.body.removeChild(slots.ref),
+        navbar: {
+            id: 'navbar',
+            getComputedStyle: () => window.getComputedStyle(slots.navbar.ref, null),
+        },
+        body: {
+            id: 'body',
+            getComputedStyle: () => window.getComputedStyle(slots.body.ref, null),
+            getAttributeName: () => document.body.getAttribute('name'),
+        },
+    };
+
+    const applications = {
+        navbar: {
+            id: 'navbar-application',
+            class: 'navbar-spa-application',
+            appendApplication: () => slots.navbar.ref.appendChild(applications.navbar.ref),
+        },
+        body: {
+            id: 'body-application',
+            class: 'body-spa-application',
+            appendApplication: () => slots.body.ref.appendChild(applications.body.ref),
+            removeApplication: () => slots.body.ref.removeChild(applications.body.ref)
+        },
+    };
+
+    const spinner = {
+        id: 'ilc-spinner',
+        class: 'ilc-spinner',
+        getRef: () => document.getElementById(spinner.id),
+    };
+
+    let clock;
+
+    beforeEach(() => {
+        window.location.hash = locationHash;
+
+        window.ilcConfig = {
+            tmplSpinner: `<div id="${spinner.id}" class="${spinner.class}">Hello! I am Spinner</div>`,
+        };
+
+        applications.navbar.ref = html`
+            <div id="${applications.navbar.id}" class="${applications.navbar.class}">
+                Hello! I am Navbar SPA application
+            </div>`;
+
+        applications.body.ref = html`
+            <div id="${applications.body.id}" class="${applications.body.class}">
+                Hello! I am Body SPA application
+            </div>`;
+
+        slots.ref = html`
```

Comment:
Could you define this property at the same time you declare `slots` variable? Now this code isn't really obvious.,